cmd/juju/action/run.go

```diff
@@ -133,53 +133,54 @@ func (c *runCommand) SetFlags(f *gnuflag.FlagSet) {
 func (c *runCommand) Info() *cmd.Info {
 	return &cmd.Info{
 		Name:    "run-action",
-		Args:    "<unit> <action name> [key.key.key...=value]",
+		Args:    "<unit> [<unit> ...] <action name> [key.key.key...=value]",
 		Purpose: "Queue an action for execution.",
 		Doc:     runDoc,
 	}
 }
 
-// Init gets the unit tag, and checks for other correct args.
+// Init gets the unit tag(s), action name and action arguments.
 func (c *runCommand) Init(args []string) error {
-	switch len(args) {
-	case 0:
+	var unitNames []string
+	for idx, arg := range args {
+		if names.IsValidUnit(arg) {
```

Comment:
So I believe the current "juju run" CLI is trying to help with the fact that your already passing an arbitrary list of things on the CLI, having 2 unbound lists makes it hard to have a concrete marker. (eg: "juju run places to run and thing to run", is ambiguous) It sounds like we'll have the same problems with 'juju run-action' since actions take parameters.
Do you find "juju run --units foo/a,foo/b" hard to use?
I feel the qualm is probably that repeating the prefix sucks, and people probably would actually write: juju run-action postgresql/{0,1} action
Which plays nicely with 'space separated' but not 'comma separated'.

Though I think we could probably support: juju run --units "pg/0 pg/1", we could space separate and comma separate. (lots of other things like --constraints are space separated anyway).

I certainly wouldn't require unifying "juju run" and "juju run-action", but I'd like to see if we can find a way to make them feel more aligned moving forward.

*My* current favorite is :
  juju run-action --units=a,b,c action

I'm happy to have a wider discussion with others on the team as to how we want the CLI to work,  I may be the only one around due to Easter holidays, but I'm certainly not the only say on CLI :).